server/src/main/java/org/opensearch/search/backpressure/settings/SearchShardTaskSettings.java

```diff
@@ -52,7 +188,96 @@ public long getTotalHeapBytesThreshold() {
         return (long) (HEAP_SIZE_BYTES * getTotalHeapPercentThreshold());
```

Comment:
The `HEAP_SIZE_BYTES` leaks here as well, so when it is `0`, both thresholds are going to be incorrect: `SearchBackpressureService::isHeapUsageDominatedBySearch` will always return `true`. I think we have to move this `getTotalHeapBytesThreshold` to `HeapTracker` as well, so the decision regarding heap tracking is made in once place only.

Reply:
How about we change the method `isHeapUsageDominatedBySearch` like this?
```
boolean isHeapUsageDominatedBySearch(List<CancellableTask> cancellableTasks, double heapPercentThreshold) {
        long usage = cancellableTasks.stream().mapToLong(task -> task.getTotalResourceStats().getMemoryInBytes()).sum();
        long threshold = (long) heapPercentThreshold * getHeapSizeBytes();
        if (isHeapTrackingSupported() && usage < threshold) {
            logger.debug("heap usage not dominated by search requests [{}/{}]", usage, threshold);
            return false;
        }
        return true;
    }
```

Reply:
:+1: This is better, thank you

Reply:
Well, I meant to keep it in `SearchBackpressureService` only. Hope that's fine?
Like, I don't see a problem in having a public getter for `HEAP_SIZE_BYTES`. Thoughts?

Reply:
The HEAP_SIZE_BYTES should be sealed: you see, your settings would have incorrect values because the knowledge that `HEAP_SIZE_BYTES` is implicit. This is not good, by sealing it and only having method to access it you free up the users of thinking that  `HEAP_SIZE_BYTES` could be incorrect.

Reply:
I did not get it. Should we have a public getter or not?

Reply:
@PritLadani public getter for `getTotalHeapPercentThreshold()` - yes, `public static long getTotalHeapBytesThreshold(double threshold )` goes to `HeapTracker`, HEAP_SIZE_BYTES becomes private

Reply:
Thanks!